airbyte-cdk/bulk/core/load/src/main/kotlin/io/airbyte/cdk/data/AirbyteValue.kt

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.cdk.data
+
+sealed interface AirbyteValue
+
+data object NullValue : AirbyteValue
+
+@JvmInline value class StringValue(val value: String) : AirbyteValue
+
+@JvmInline value class BooleanValue(val value: Boolean) : AirbyteValue
+
+@JvmInline value class IntegerValue(val value: Long) : AirbyteValue
+
+@JvmInline value class NumberValue(val value: Double) : AirbyteValue
+
+@JvmInline value class DateValue(val value: String) : AirbyteValue
+
+@JvmInline value class TimestampValue(val value: String) : AirbyteValue
```

Comment:
should we have separate classes for time/timestamp with/without tz?
```kotlin
class TimeWithTimezone(OffsetTime)
class TimeWithoutTimezone(LocalTime)
class TimestampWithTimezone(OffsetDateTime)
class TimestampWithoutTimezone(LocalDateTime)
```

Reply:
I've gone back and forth on it. Only if we're converting timestamps I think?

Reply:
:shrug:

let's do that for now, and if we run into trouble, we can always switch this around